src/Build/BuildCheck/Analyzers/NoEnvironmentVariablePropertyAnalyzer.cs

```diff
@@ -26,13 +27,15 @@ internal sealed class NoEnvironmentVariablePropertyAnalyzer : BuildAnalyzer
     private readonly HashSet<EnvironmentVariableIdentityKey> _environmentVariablesReported = new HashSet<EnvironmentVariableIdentityKey>();
 
     private bool _isVerboseEnvVarOutput;
+    private EvaluationAnalysisScope _scope;
 
     public override string FriendlyName => "MSBuild.NoEnvironmentVariablePropertyAnalyzer";
 
     public override IReadOnlyList<BuildAnalyzerRule> SupportedRules { get; } = [SupportedRule];
 
     public override void Initialize(ConfigurationContext configurationContext)
     {
+        _scope = configurationContext.BuildAnalyzerConfig[0].EvaluationAnalysisScope;
```

Comment:
can BuildAnalyzerConfig ever be null?

Reply:
No - that is the beauty of the nullability annotations :-)

But if you have asked if the array can be empty - then this cannot be enforced on the code analysis level, but still - our infra allways sets this to array of proper size (based on the rules count)